### PR TITLE
chore(Popover): deprecate `variant` and `customIcon` props

### PR DESCRIPTION
**Pull request by HeartSquared**
## What

Deprecate variant for Popover.
Also fixed the RTL styles.

## Why

[KZN-2528](https://cultureamp.atlassian.net/jira/software/c/projects/KZN/boards/634?selectedIssue=KZN-2528)
Part of the a11y contrast improvements initiative.

[KZN-2528]: https://cultureamp.atlassian.net/browse/KZN-2528?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/components/src/Popover/Popover.module.scss [hunk 1]
```diff
@@ -127,7 +127,7 @@ $cautionary-box-border-color: var(--color-yellow-700);
   display: flex;
   align-items: center;
   white-space: nowrap;
-  padding-right: $spacing-md;
+  padding-inline-end: $spacing-md;
```
Comment (mcwinter07) [inline packages/components/src/Popover/Popover.module.scss:130]: Nice, could potentially update this to spacing-24, or even the css var if you're keen :)

**Review by mcwinter07 — COMMENTED** [review]
Should we have a codemod as part of this release or is that a separate task we want to do?

**Comment by HeartSquared** [comment]
> Should we have a codemod as part of this release or is that a separate task we want to do?

Oh good point. I suppose it does make sense to have a codemod that removes the props.

Looks like `customIcon` is never used by anyone either, so it would be pretty safe!
https://github.com/search?q=org%3Acultureamp%20customIcon&type=code

**Review by mcwinter07 — APPROVED** [review]
Nice abstraction of the `removeProps` transformer

**Pull request merged**